### DISCUSSION TITLE
"Previous" Exceptions

### DISCUSSION THREAD

**Discussion by PHLAK** [spatie/flareapp.io-roadmap#44] [Ideas]
As far as I can tell when an exception is thrown (and sent to Flare) that contains a "previous" exception there's no way to view the previous exception from the Flare UI. 

For example, given the following

```php
try {
    // Something that throws an exception
} catch (SomeException $exception) {
    throw AnotherException('Message', 0, $exception); // Note the third parameter here
}
```

there's no way to access/view the contents (e.g. message, code, etc. ) of `SomeExcption`.

For reference you can access the previous exception (and it's related data) through the [`getPrevious()`](https://www.php.net/manual/en/exception.getprevious.php) method on an Exception class.

```php
$someException = new Exception('First exception');

$anotherException = new Exception('Second exception', 0, $someException);

$anotherException->getPrevious()->getMessage(); // Returns 'First exception'
```

It would be extremely useful to be able to access the previous exception via the UI either in the Stack trace tab or via a dedicated tab.

**Comment by AlexVanderbist** [comment]
Hi @PHLAK, thanks for the suggestion! This is one of the things we're currently working on. Looking forward to sharing more soon :)

**Reply by PHLAK** [reply]
Awesome, thank you!

**Reply by Naugrimm** [reply]
Hi @AlexVanderbist and the Flare team,

I was wondering if there's been any progress on implementing the Previous-Exceptions-feature? This would be really helpful for the debugging workflow when using nested exceptions.

Thanks for all your work on Flare!

**Reply by AlexVanderbist** [reply]
Hey, if I remember correctly, this became more complicated than we anticipated, especially since Laravel has decided to drop support for this in their new error page. Do you have a specific use case where this might still be useful?

**Reply by Naugrimm** [reply]
We wrap the original exception to add business context while preserving the technical details:

```php
try {
    $newPlan = $this->newPlan($event);
} catch (ModelNotFoundException $exception) {
    throw ChangeEndcustomerFailed::make($event, $exception);
}
```

The `$this->newPlan()` method can throw Laravel's `ModelNotFoundException` in multiple scenarios. Since we cannot directly modify the `ModelNotFoundException` to include our business context, we wrap it in our own custom exception. However, it's crucial to maintain access to the original `ModelNotFoundException` and its stack trace to understand exactly where and why the model lookup failed.

This pattern allows us to:
1. Add business-level context about what operation was being attempted
2. Preserve the technical details of why the operation failed
3. Maintain the original stack trace for debugging purposes

Of course, we do have the information in the local error log. Nevertheless it would be really helpful to see it directly in Flare, too.

**Reply by GregoireGaonach** [reply]
We have exactly the same pattern in our app, where we wrap our critical paths exceptions in domain-level exceptions with business context so it can fail gracefully. However, Flare only shows the top-level exception, which makes debugging much harder. We would love to see this feature added.

**Reply by AlexVanderbist** [reply]
Hey, thanks for the feedback! I can see how this would be useful when you're wrapping exceptions to add business context while preserving the technical details.

We've recently added this back to our roadmap. You should see the implementation show up in the `spatie/flare-client-php` package soon. However, I can't give an ETA for the UI changes on Flare itself yet. I'll update this issue once we have a more refined plan.

**Reply by GregoireGaonach** [reply]
Thanks a lot @AlexVanderbist ; for the moment we add some details about the previous exception using `context`. The team and I can't wait <3
